core/src/com/unciv/ui/worldscreen/WorldScreen.kt

```diff
@@ -363,6 +364,24 @@ class WorldScreen(val gameInfo: GameInfo, val viewingCiv:CivilizationInfo) : Bas
         }
     }
 
+    /** @return Whether a unit's past movements should be visible to the player. */
+    private fun isUnitPastVisible(unit: MapUnit): Boolean {
```

Comment:
Why is this function in the world screen and not in the holder?

Reply:
It uses `WorldScreen.viewingCiv`, and it requires `.gameInfo` where it's used since it filters through all units… It needs the same data/properties as `WorldScreen`? Idk. It just didn't feel like a good fit for the map holder to me— Deciding what information to expose is too close to game rules instead of just the map display— Though looking at WorldScreen.kt again, I think I was mistaken to think that's a better place for it…

…Hm. I've moved it into a separate helper class, to keep all the "Is this visualization visible?" logic distinct, à la UnitMovementAlgorithms.